### PR TITLE
backport: bitcoin#17428, #17785, #19109, #19610, #19804, #19879, #20022, #20027, #20090, #20131

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Just regular bitcoin backports, mostly from v21

## What was done?
 - bitcoin/bitcoin#19610
 - bitcoin/bitcoin#19879
 - bitcoin/bitcoin#17785
 - bitcoin/bitcoin#19804
 - bitcoin/bitcoin#20022
 - bitcoin/bitcoin#20027
 - bitcoin/bitcoin#20131
 - bitcoin/bitcoin#17428
 - bitcoin/bitcoin#20090
 - bitcoin/bitcoin#19109

## Note for reviewers:
 - bitcoin#20027 also fixes previous merge error in `test/functional/wallet_resendwallettransactions.py` with duplicated code
 - bitcoin#17428 used to be merged before partially, see #5574 which exclude commit https://github.com/bitcoin/bitcoin/commit/0a85e5a7bc8dc6587963e2e37ac1b087a1fc97fe

 
## How Has This Been Tested?
Run unit/functional tests

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -149,10 +151,19 @@ static constexpr std::chrono::seconds AVG_ADDRESS_BROADCAST_INTERVAL{30};
...
  *  Masternode outbound peers get quarter this delay. */
 static const unsigned int INVENTORY_BROADCAST_INTERVAL = 5;
-/** Maximum number of inventory items to send per transmission.
+/** Maximum rate of inventory items to send per second.
  *  Limits the impact of low-fee transaction floods.
  *  We have 4 times smaller block times in Dash, so we need to push 4 times more invs per 1MB. */
-static constexpr unsigned int INVENTORY_BROADCAST_MAX_PER_1MB_BLOCK = 4 * 7 * INVENTORY_BROADCAST_INTERVAL;
+static constexpr unsigned int INVENTORY_BROADCAST_PER_SECOND = 7;
+/** Maximum number of inventory items to send per transmission. */
+static constexpr unsigned int INVENTORY_BROADCAST_MAX_PER_1MB_BLOCK = 4 * INVENTORY_BROADCAST_PER_SECOND * INVENTORY_BROADCAST_INTERVAL;
```
Comment (UdjinM6) [inline src/net_processing.cpp:159]: 19109: should probably adjust `INVENTORY_BROADCAST_PER_SECOND` instead, not sure
```suggestion
static constexpr unsigned int INVENTORY_BROADCAST_PER_SECOND = 7 * 4;
/** Maximum number of inventory items to send per transmission. */
static constexpr unsigned int INVENTORY_BROADCAST_MAX_PER_1MB_BLOCK = INVENTORY_BROADCAST_PER_SECOND * INVENTORY_BROADCAST_INTERVAL;
```

Comment (knst) [reply src/net_processing.cpp:159]: we have a strange name `INVENTORY_BROADCAST_MAX_PER_1MB_BLOCK` to be sure that multiplier `4` is not lost. If change it like in suggestion - we need to rename `INVENTORY_BROADCAST_PER_SECOND` then.

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

File: src/rpc/net.cpp [hunk 2]
```diff
@@ -118,7 +127,9 @@ static UniValue getpeerinfo(const JSONRPCRequest& request)
                     {RPCResult::Type::STR, "subver", "The string version"},
                     {RPCResult::Type::BOOL, "inbound", "Inbound (true) or Outbound (false)"},
                     {RPCResult::Type::BOOL, "addnode", "Whether connection was due to addnode/-connect or if it was an automatic/inbound connection"},
-                    {RPCResult::Type::STR, "connection_type", "Type of connection: \n" + Join(CONNECTION_TYPE_DOC, ",\n") + "."},
+                    {RPCResult::Type::STR, "connection_type", "Type of connection: \n" + Join(CONNECTION_TYPE_DOC, ",\n") + ".\n"
```
Comment (thephez) [inline src/rpc/net.cpp:130]: This is another case where the help output order doesn't align with the actual RPC response. In the actual output, this field comes last: https://github.com/dashpay/dash/blob/develop/src/rpc/net.cpp#L256

**Pull request merged**